File: src/ERP/acme.atena.domain/DTO/Product/DevolucaoCompra.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace acme.atena.domain.DTO.Product
{
    public class DevolucaoCompra : AbstractEntityCompetencia
    {
        public string Motivo { get; set; }
        public Guid CompraProdutoId { get; set; }
        public Guid ProdutoId { get; set; }
        public bool IsParcial { get; set; } = false;

        public virtual CompraProduto CompraProduto { get; set; }
        public virtual Produto Produto { get; set; }
    }
}

File: site/coreui-free-angular-admin-template-master/src/app/view-model/response/product/devolucao-compra.ts
import { Produto } from './produto';
import { AbstractEntityCompetencia } from "../abstract-entity-competencia";
import { CompraProduto } from "./compra-produto";

export class DevolucaoCompra extends AbstractEntityCompetencia{
  public  motivo:string;
  public  compraProdutoId:string;
  public  produtoId:string;
  public  isParcial:boolean;

  public  compraProduto: CompraProduto;
  public  produto: Produto;
}
